disvae/losses.py

```diff
@@ -229,3 +325,32 @@ def _kl_normal_loss(mean, logvar, storer=None):
         for i in range(latent_dim):
             storer['kl_loss_' + str(i)] += latent_kl[i].item()
     return total_kl
+
+
+def _permute_dims(latent_dist):
+    """
+    Implementation of Algorithm 1 in ref [1]. Randomly permutes the sample from
+    q(z) (latent_dist) across the batch for each of the latent dimensions (mean
+    and log_var).
+
+    Parameters
+    ----------
+    latent_dist : torch.Tensor
+        sufficient statistics of the latent dimension. E.g. for gaussian
+        (mean, log_var), each of shape : (batch_size, latent_dim).
+
+    References :
+        [1] Kim, Hyunjik, and Andriy Mnih. "Disentangling by factorising."
+        arXiv preprint arXiv:1802.05983 (2018).
+
+    """
+
+    perm = torch.zeros(latent_dist.size())
+    dim_j, batch_size, dim_z = latent_dist.size()
+
+    for j in range(dim_j):
+        for z in range(dim_z):
```

Comment:
IMPORTANT : you seem to be permuting the latent_dist (i.e mean and variance) while in the paper (and in 1konny) they permute the samples!!!!!!! (that's why you have one more loop.
I'm pretty sure this is not the same. One thing that might be the same (i haven't though about it too much yet) is if you use the same permutation for the mean and the variance. But it also shows a fudamental difference between you code and the paper: you are actually giving the distribution to the discriminator not z !! FOr example you say `z_perm = _permute_dims(latent_dist)` but it's actually `latent_dist_perm = _permute_dims(latent_dist)`. I'd have to think abou tit more carefully if that is also correct.
Are you able to get good results?

Reply:
Thanks for pointing this out. I'll fix the error!